src/hooks/auth0-provider.js

```diff
@@ -0,0 +1,87 @@
+import React, {useReducer, useState} from 'react';
+import {useCallback, useMemo} from 'react';
+import jwt_decode from 'jwt-decode';
+import PropTypes from 'prop-types';
+import Auth0Context from './auth0-context';
+import Auth0 from '../auth0';
+import reducer from './reducer';
+import {idTokenNonProfileClaims} from '../jwt/utils';
+
+const initialState = {
+  isLoading: true,
+  user: null,
+  error: null,
+};
+
+/**
+ * @ignore
+ */
+const getIdTokenProfileClaims = idToken => {
+  const payload = jwt_decode(idToken);
+
+  const profileClaims = Object.keys(payload).reduce((profile, claim) => {
+    if (!idTokenNonProfileClaims.includes(claim)) {
+      profile[claim] = payload[claim];
+    }
+
+    return profile;
+  }, {});
+
+  return profileClaims;
+};
+
+const Auth0Provider = ({domain, clientId, children}) => {
+  const [client] = useState(new Auth0({domain, clientId}));
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const authorize = useCallback(
+    async (...options) => {
+      try {
+        const credentials = await client.webAuth.authorize(...options);
+        const claims = getIdTokenProfileClaims(credentials.idToken);
+        dispatch({type: 'LOGIN_COMPLETE', user: claims});
+        return credentials;
```

Comment:
I assume that `authorize` wont return anything once you add credentials manager/getAccessToken to the hook?

Reply:
Correct, there'll be some updates once the credential manager work comes in.